File: java/app/src/main/java/net/fbvictorhugo/j/barreirasanitaria/ui/adapter/PessoasRecyclerAdapter.java
package net.fbvictorhugo.j.barreirasanitaria.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import net.fbvictorhugo.j.barreirasanitaria.R;
import net.fbvictorhugo.j.barreirasanitaria.data.model.Pessoa;

import java.util.List;

import androidx.appcompat.widget.AppCompatTextView;
import androidx.recyclerview.widget.RecyclerView;

public class PessoasRecyclerAdapter extends RecyclerView.Adapter<PessoasRecyclerAdapter.MyViewHolder> {

    private final Context mContext;
    private List<Pessoa> mDataset;
    private OnItemClickListener mOnItemClickListener;

    public PessoasRecyclerAdapter(final Context context) {
        mContext = context;
    }

    @Override
    public PessoasRecyclerAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.adapter_lista_padrao, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, final int position) {
        holder.textViewPrincipal.setText(mDataset.get(position).getNome());
        String textoCpf = String.format(mContext.getResources().getString(R.string.texto_cpf_), String.valueOf(mDataset.get(position).getNumeroDocumento()));
        holder.textViewSecundario.setText(textoCpf);
        holder.itemView.setOnClickListener(view -> {
            if (mOnItemClickListener != null) {
                mOnItemClickListener.onClick(mDataset.get(position));
            }
        });
        holder.itemView.setOnLongClickListener(view -> {
            if (mOnItemClickListener != null) {
                mOnItemClickListener.onLongClick(mDataset.get(position));
            }
            return true;
        });

    }

    @Override
    public int getItemCount() {
        return mDataset != null ? mDataset.size() : 0;
    }

    public void adiciona(Pessoa pessoa) {
        if (mDataset != null) {
            mDataset.add(pessoa);
            notifyDataSetChanged();
        }
    }

    public void atualiza(List<Pessoa> newDataset) {
        mDataset = newDataset;
        notifyDataSetChanged();
    }

    static class MyViewHolder extends RecyclerView.ViewHolder {

        AppCompatTextView textViewPrincipal;
        AppCompatTextView textViewSecundario;

        MyViewHolder(View baseView) {
            super(baseView);
            textViewPrincipal = baseView.findViewById(R.id.adapter_item_tv_principal);
            textViewSecundario = baseView.findViewById(R.id.adapter_item_tv_sencundario);
        }
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        mOnItemClickListener = listener;
    }

    public interface OnItemClickListener {

        void onClick(Pessoa pessoa);

        void onLongClick(Pessoa pessoa);
    }

}

File: kotlin/app/src/main/java/net/fbvictorhugo/k/barreirasanitaria/ui/adapter/PessoasRecyclerAdapter.kt
package net.fbvictorhugo.k.barreirasanitaria.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatTextView
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.adapter_lista_padrao.view.*
import net.fbvictorhugo.k.barreirasanitaria.R
import net.fbvictorhugo.k.barreirasanitaria.data.model.Pessoa

class PessoasRecyclerAdapter(private val _context: Context) :
    RecyclerView.Adapter<PessoasRecyclerAdapter.MyViewHolder>() {

    private var _dataset: ArrayList<Pessoa> = ArrayList()
    var onItemClickListener: OnItemClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view =
            LayoutInflater.from(_context).inflate(R.layout.adapter_lista_padrao, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.textViewPrincipal.text = _dataset[position].nome
        val textoCpf = String.format(
            _context.resources.getString(R.string.texto_cpf_), _dataset[position].numeroDocumento
        )
        holder.textViewSecundario.text = textoCpf
        holder.itemView.setOnClickListener { onItemClickListener?.onClick(_dataset[position]) }
        holder.itemView.setOnLongClickListener {
            onItemClickListener?.onLongClick(_dataset[position])
            true
        }

    }

    override fun getItemCount(): Int {
        return _dataset.size
    }

    fun atualiza(dataset: ArrayList<Pessoa>) {
        _dataset = dataset
        notifyDataSetChanged()
    }

    class MyViewHolder(baseView: View) : RecyclerView.ViewHolder(baseView) {
        val textViewPrincipal: AppCompatTextView = baseView.adapter_item_tv_principal
        val textViewSecundario: AppCompatTextView = baseView.adapter_item_tv_sencundario
    }

    interface OnItemClickListener {
        fun onClick(pessoa: Pessoa)
        fun onLongClick(pessoa: Pessoa)
    }

}
